test/HalfStorage/vloada_half2_global.cl

```diff
@@ -10,7 +10,7 @@
 
 kernel void foo(global float2* A, global uint* B, uint n) {
   A[0] = vloada_half2(n, (global half*)B);
-  A[1] = vloada_half2(0, (global half*)B+2);
+  A[1] = vloada_half2(0, (global half*)(B+1));
```

Comment:
This seems strange. Why is the checking code now correct for a different address?

Reply:
The cast is applied before the add, so the address is the same, the difference is that before we used to have a pointer addition with a half type. This leads to a complete change in the generated code with half types all around.
With this new form, we are doing the pointer addition with a uint type.

Reply:
Ah I missed the parentheses.